digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/linearasset/manoeuvre/ManoeuvreDAO.scala

```diff
@@ -372,20 +397,29 @@ class ManoeuvreDao() extends PostGISLinearAssetDao{
       """.as[Long].first
   }
   
-  def insertSamuutusChange(row:Seq[ChangedManoeuvre]): Unit ={
-    val insert= "insert into manouvre_samuutus_work_list (assetId,linkIds) values ((?),(?)) ON CONFLICT (assetId) do nothing "
-    MassQuery.executeBatch(insert){p=>
-      row.foreach(a=>{
-        p.setLong(1,a.manoeuvreId)
-        p.setString(2,a.linkIds.map(t=>s"'$t'").mkString(","))
+  def insertManoeuvreToWorkList(manoeuvresToInsert:Seq[Manoeuvre]): Unit ={
+    val insert= "insert into manouvre_samuutus_work_list (assetId, linkIds, exception_types, validity_periods, additional_info, created_date) values ((?), (?), (?), (?), (?), current_timestamp) ON CONFLICT (assetId) do nothing "
+    MassQuery.executeBatch(insert) { p =>
+      manoeuvresToInsert.foreach(manoeuvre => {
+        val linkIds = manoeuvre.elements.flatMap(elem => Seq(elem.sourceLinkId, elem.destLinkId)).toSet
+        val exceptionTypesArray = p.getConnection.createArrayOf("INTEGER", manoeuvre.exceptions.map(_.asInstanceOf[AnyRef]).toArray)
+        val validityPeriodsStrings = manoeuvre.validityPeriods.map(period => s"${period.days.value}-${period.startHour}-${period.endHour}-${period.startMinute}-${period.endMinute}")
+        val validityPeriodsArray = p.getConnection.createArrayOf("TEXT", validityPeriodsStrings.toArray.map(_.asInstanceOf[AnyRef]))
```

Comment:
mitä tämä .getConnection.  tekee?

Reply:
Haetaan tuon PreparedStatementin kantayhteys. Tuo .createArrayOf metodi on Connection luokan metodi, joten yhteys pitää hakea sitä varten. Ei ole syvällisempää tietoa miksi toimii näin